helm-chart/splunk-kubernetes-logging/templates/_helpers.tpl

```diff
@@ -133,8 +124,135 @@ Create the name of the service account to use
 {{- printf "%s/%s:%s" .Values.image.registry .Values.image.name .Values.image.tag -}}
 {{- end -}}
+
+{{/*
+Converts a path glob to a regular expression.
+*/}}
+{{- define "splunk-kubernetes-logging.glob_to_regex" -}}
+  {{- $str := . -}}
+  {{- $str := $str | replace "\\" "\\\\" -}}
+  {{- $str := $str | replace "." "\\." -}}
+  {{- $str := $str | replace "*" ".*" -}}
+  {{- $str := $str | replace "?" "\\?" -}}
+  {{- $str := $str | replace "{" "\\{" -}}
+  {{- $str := $str | replace "}" "\\}" -}}
+  {{- $str := $str | replace "(" "\\(" -}}
+  {{- $str := $str | replace ")" "\\)" -}}
+  {{- $str := $str | replace "[" "\\[" -}}
+  {{- $str := $str | replace "]" "\\]" -}}
+  {{- $str := $str | replace "^" "\\^" -}}
+  {{- $str := $str | replace "$" "\\$" -}}
+  {{- print "^" $str "$" -}}
+{{- end }}
+
+{{/*
+Generates the jq_transformer query to ensure each record has the appropriate index.
+*/}}
+{{- define "splunk-kubernetes-logging.index_jq_filter" -}}
+  {{- if (or .Values.splunk.hec.indexRouting .Values.global.splunk.hec.indexRouting) -}}
+    {{- include "splunk-kubernetes-logging.index_jq_filter_implicit" . -}}
+  {{- else -}}
+    {{- include "splunk-kubernetes-logging.index_jq_filter_explicit" . -}}
+  {{- end -}}
+{{- end -}}
+
+{{/*
+Generates the jq_transformer query to ensure each record has the appropriate index when routing by namespace.
+*/}}
+{{- define "splunk-kubernetes-logging.index_jq_filter_implicit" -}}
+  {{- $splunkIndexRoutingDefault := or .Values.splunk.hec.indexRoutingDefaultIndex .Values.global.splunk.hec.indexRoutingDefaultIndex "main" -}}
+  {{- print ".record | .index = (if .namespace == \"default\" then " ($splunkIndexRoutingDefault | quote) " else .namespace//" ($splunkIndexRoutingDefault | quote) " end)" -}}
+{{- end -}}
+
+{{/*
+Generates the jq_transformer query to ensure each record has the appropriate index when not routing by namespace.
+*/}}
+{{- define "splunk-kubernetes-logging.index_jq_filter_explicit" -}}
+  {{- $journalPath := or .Values.journalLogPath .Values.global.journalLogPath "/run/log/journal/" -}}
+  {{- $splunkHecIndex := (or .Values.splunk.hec.indexName .Values.global.splunk.hec.indexName) -}}
+  {{- $logDefsByIndexTag := dict -}}
+  {{- range $name, $logDef := .Values.logs -}}
+    {{- if (hasKey $logDef "index") -}}
+      {{- $index := or $logDef.index $splunkHecIndex -}}
+      {{- if (ne $index $splunkHecIndex) -}}
+        {{- $sourcetype := or $logDef.sourcetype $name -}}
+        {{- $tag := dict "value" "" -}}
+
+        {{- if $logDef.from.pod -}}
+          {{- $_ := set $tag "value" "tail.containers.*" -}}
+        {{- else if $logDef.from.file -}}
+          {{- $_ := set $tag "value" (print "tail.file." $sourcetype) -}}
+        {{- else if $logDef.from.journald -}}
+          {{- $_ := set $tag "value" (print "journald." $sourcetype) -}}
+        {{- else -}}
+          {{- print "def TODO_support_log_" $name ": " ($logDef | toJson) "; " -}}
```

Comment:
I am not sure of the purpose of this "def TODO_support_log_" is it for documentation?

Reply:
This line is not for documentation, but rather is for debugging when an unsupported log type is encountered.